src/datetime_rotation.jl

```diff
@@ -28,19 +33,25 @@ logger = DatetimeRotatingFileLogger(log_dir, raw"\a\c\c\e\s\s-yyyy-mm-dd.\l\o\g"
 logger = DatetimeRotatingFileLogger(log_dir, raw"yyyy-mm-dd-HH.\l\o\g") do io, args
     println(io, args.level, " | ", args.message)
 end
+
+# Example callback function to compress the recently-closed file
+callback(file) = run(`gzip $(file)`)
 """
 mutable struct DatetimeRotatingFileLogger <: AbstractLogger
     logger::Union{SimpleLogger,FormatLogger}
     dir::String
     filename_pattern::DateFormat
     next_reopen_check::DateTime
     always_flush::Bool
+    reopen_lock::ReentrantLock
+    current_file::Union{String,Nothing}
```

Comment:
What do you think of instead just reading the filename out of the steam before we close it?

I think it is too nasty,
though it is an option
```
julia> k = open(abspath("src/threading.h"))
IOStream(<file /Users/oxinabox/JuliaEnvs/julia/src/threading.h>)

julia> k.name
"<file /Users/oxinabox/JuliaEnvs/julia/src/threading.h>"
```

Reply:
IMO that is a bit iffy since we rely on an undocumented(?) internal field, and also need to extract the file from that string.

Reply:
fair